app/frontend/components/common/loadingOverlay.tsx

```diff
@@ -2,16 +2,24 @@ import {h} from 'preact'
 import {useSelector} from '../../helpers/connect'
 
 const LoadingOverlay = () => {
+  const toArray = (message: string | string[]) =>
+    message ? (Array.isArray(message) ? message : [message]) : ['']
+
   const {loading, loadingMessage} = useSelector((state) => ({
-    loadingMessage: state.loadingMessage,
+    loadingMessage: toArray(state.loadingMessage),
```

Comment:
I'd consider not overcomplicating this with extending the message to be an array of lines (which is kind of unexpected from the name and creeps into the rest of the codebase in form of extended definitions) and I'd just add a functionality to parse newline characters in the string passed and split it into proper lines at presentation. 

The line list IMHOof makes the API more bloated than it needs to be in the vast majority of cases and a "proper" abstraction, i.e. "messageLines" instead of messages seems like an overkill for one special case. If we ever need to refactor this, it should be rather straightforward to figure out what messages are multiline by checking where the component is instantiated

https://stackoverflow.com/questions/35351706/how-to-render-a-multi-line-text-string-in-react

Reply:
Refactored